Providers/Profile.swift

```diff
@@ -485,36 +485,23 @@ open class BrowserProfile: Profile {
         return self.remoteClientsAndTabs.insertOrUpdateTabs(tabs)
     }
 
-    public func sendItem(_ item: ShareItem, toClients clients: [RemoteClient]) -> Success {
-        func clientForRemoteDevice(_ remoteDevice: RemoteDevice) -> RemoteClient? {
-            return clients.find({ $0.fxaDeviceId == remoteDevice.id })
-        }
-
+    public func sendItem(_ item: ShareItem, toDevices devices: [RemoteDevice]) -> Success {
         guard let account = self.getAccount() else {
             return deferMaybe(NoAccountError())
         }
 
         let scratchpadPrefs = self.prefs.branch("sync.scratchpad")
         let id = scratchpadPrefs.stringForKey("clientGUID") ?? ""
         let command = SyncCommand.displayURIFromShareItem(item, asClient: id)
-        let fxaDeviceIds = clients.compactMap { $0.fxaDeviceId }
-
-        // If FxA Messages (Pushbox) is not enabled for this build, simply send the
-        // tabs using the old mechanism via Sync.
-        guard AppConstants.MOZ_FXA_MESSAGES else {
-            return self.remoteClientsAndTabs.insertCommands([command], forClients: clients) >>> {
-                self.syncManager.syncClients() >>> {
-                    account.notify(deviceIDs: fxaDeviceIds, collectionsChanged: ["clients"], reason: "sendtab")
-                    return succeed()
-                }
-            }
-        }
+        let fxaDeviceIds = devices.compactMap { $0.id }
 
         let result = Success()
 
-        self.remoteClientsAndTabs.getRemoteDevices() >>== { remoteDevices in
-            let newRemoteDevices = remoteDevices.filter({ fxaDeviceIds.contains($0.id ?? "") && account.commandsClient.sendTab.isDeviceCompatible($0) })
-            var oldRemoteClients = remoteDevices.filter({ fxaDeviceIds.contains($0.id ?? "") && !account.commandsClient.sendTab.isDeviceCompatible($0) }).compactMap({ clientForRemoteDevice($0) })
+        self.remoteClientsAndTabs.getClients() >>== { clients in
+            let newRemoteDevices = devices.filter({ fxaDeviceIds.contains($0.id ?? "") && account.commandsClient.sendTab.isDeviceCompatible($0) })
+            var oldRemoteClients = devices.filter({ fxaDeviceIds.contains($0.id ?? "") && !account.commandsClient.sendTab.isDeviceCompatible($0) }).compactMap({ remoteDevice in
```

Comment:
We can probably remove the additional `fxaDeviceIds.contains($0.id ?? "")` check here again since `fxaDeviceIds` is derived from `devices` already.